src/sentry/static/sentry/app/views/organizationIntegrations/integrationListDirectory.tsx

```diff
@@ -289,6 +319,26 @@ export class OrganizationIntegrations extends AsyncComponent<
     });
   };
 
+  onCategorySelect = async ({value: category}: {value: string}) => {
+    this.setState({selectedCategory: category}, () => {
+      if (!category) {
+        return this.setState({displayedList: this.state.list});
+      }
+      const result = this.state.list.filter(integration => {
+        if (isSentryApp(integration)) {
+          const features = ['internal', 'unpublished'].includes(integration.status)
+            ? [{featureGate: integration.status, description: ''}]
+            : integration.featureData;
+          return getCategories(features).includes(category);
+        }
+        if (isPlugin(integration)) {
+          return getCategories(integration.featureDescriptions).includes(category);
+        }
+        return getCategories(integration.metadata.features).includes(category);
```

Comment:
I'm not sure what the refactor would look like, but there's a lot of duplicate logic here. You could try something like:
```js
let features;
if (isSentryApp(integration) {
  features = // that ternary
else if (isPlugin(integration) {
  features = integration.featureDescriptions
} else {
  features = integration.metadata.features
}

return getCategories(features).includes(category)
```

Reply:
It might also apply in the `.reduce` above

Reply:
@leeandher I don't think it's a meaningful enough reduction in complexity to do that refactor.

Reply:
@NisanthanNanthakumar I agree with @leeandher that some refactoring here would be an improvement. You forgot about the fact that we have this logic duplicated in `abstractIntegrationDetailedView.tsx` lol. I think you should make a function called `getCategoriesFromIntegration` that takes an integration of type `AppOrProviderOrPlugin` as an argument and returns a list of categories.

Reply:
@scefali actually, we don't use this logic in `abstractIntegrationDetailedView.tsx`. We use `getCategories` from `integrationUtil.tsx` and it uses the `this.featureData` method that gets implemented by the child classes.

Reply:
@NisanthanNanthakumar My bad, I meant `integrationListDirectory.tsx`. The line `['internal', 'unpublished'].includes(app.status)` appears there.

Reply:
@scefali Ok, so I agree that 
```
const features = ['internal', 'unpublished'].includes(integration.status)
            ? [{featureGate: integration.status, description: ''}]
            : integration.featureData;
```
is duplicated many times. I think it's work separating this out into a separate function `getSentryAppCategories`.

Reply:
@NisanthanNanthakumar ok but I think we should still have a function called `getCategoriesFromIntegration` that will work on every type of integration